packages/gateway/src/gateway.js

```diff
@@ -77,18 +85,36 @@ export async function gatewayGet(request, env, ctx) {
     // forward winner gateway response
     return winnerGwResponse.response
   } catch (err) {
+    let responses
+    // We can already settle requests if Aggregate Error, as all promises were already rejected
+    if (err instanceof AggregateError) {
+      responses = await pSettle(gatewayReqs)
+    }
+
     ctx.waitUntil(
       (async () => {
         // Update metrics as all requests failed
-        const responses = await pSettle(gatewayReqs)
+        const gatewayResponses = responses || (await pSettle(gatewayReqs))
         await Promise.all(
-          responses.map((r) =>
+          gatewayResponses.map((r) =>
             updateGatewayMetrics(request, env, r.value, false)
           )
         )
       })()
     )
 
+    // Redirect if all failed and at least one gateway was rate limited
```

Comment:
I think we only want to redirect when all the gateways are rate limiting us (or we prevented the request because it would cause rate limiting)...because we're literally not able to service the request. If we got error responses from everyone is probably because the content could not be retrieved OR a problem with the content (not a unixfs thing). It's unlikely that all gateways will be down at the same time and in these cases we should not redirect and get the user to send another request because it's unlikely it'll work, OR all the gateways are on 🔥  and they don't need extra traffic.

Reply:
Yes, if we have one failure that was not rate limit related we should not redirect!

But, there is an edge case here that I was including. For instance: What if we prevent two requests from running and the other is rate limited, or better, if we prevent all requests. So, what we need here is a conditional with:

All response status are `HTTP_STATUS_RATE_LIMITED` OR we prevented the request from happening